common/quotas/ratelimiter.go

```diff
@@ -126,19 +126,28 @@ func (rl *RateLimiter) shouldUpdate(maxDispatchPerSecond *float64) bool {
 	}
 }
 
-type dynamicRateLimiter struct {
+// DynamicRateLimiter implements a dynamic config wrapper around the rate limiter
+type DynamicRateLimiter struct {
 	rps RPSFunc
 	rl  *RateLimiter
 }
 
 // NewDynamicRateLimiter returns a rate limiter which handles dynamic config
-func NewDynamicRateLimiter(rps RPSFunc) Policy {
+func NewDynamicRateLimiter(rps RPSFunc) *DynamicRateLimiter {
 	initialRps := rps()
 	rl := NewRateLimiter(&initialRps, _defaultRPSTTL, int(rps()))
-	return &dynamicRateLimiter{rps, rl}
+	return &DynamicRateLimiter{rps, rl}
 }
 
-func (d *dynamicRateLimiter) Allow() bool {
```

Comment:
`dynamicRateLimiter` is different than `DomainRateLimitPolicy`
`dynamicRateLimiter` should not really require any input information other than perhaps how many token to be consumed.

one more thing, `DomainRateLimitPolicy` is not a policy (which should only contains configurations)
perhaps the `DomainRateLimitPolicy` could be better named something like multi-stage-rate-limiter

Reply:
I like that.. that's a better name..